### PR TITLE
Validate/spec

### PR DESCRIPTION
**Pull request by Rishi-Mishra0704**
# Add validations and specs for models

- Add the following validations:
    1. For the User model:
          - [x]         Name must not be blank.
          - [x]         PostsCounter must be an integer greater than or equal to zero.
    2.  For the Post model
          - [x]          Title must not be blank.
          - [x]          Title must not exceed 250 characters.
          - [x]          CommentsCounter must be an integer greater than or equal to zero.
          - [x]          LikesCounter must be an integer greater than or equal to zero.

- Add the following unit tests
  1. Add spec for user model
  2. Add spec for post model
  3. Add spec for comment model
  4. Add spec for like model

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      post.title = nil
+      expect(post).to_not be_valid
+    end
+
+    it 'Invalid if more than 250 characters' do
+      post = Post.create(
+        title: 'sdghjashgdjahsgjdahsgjdahgsjdsgjashdgjshgjash
+        gdjashgdjashgddfdjhkjdhkjdhfkjdhksjdhfyujashgdjashgd
+        jashgdjashgvcbvndasdasdasdadvcvbnmeurioeuroeirueoirueoriue
+        ordjashdgasjhdgjashdgjashdgagellodfdhjdhkfjoeimamsdhst
```
Comment (TadesseDev) [inline spec/models/post_spec.rb:24]: - [ ] Dear @Rishi-Mishra0704, You have done a great job adding the tests, but please remember we need to test all our custom methods in our model, as part of this, Please add a test for your `recent_five_comments` method inside your `Post` model.

File: spec/models/like_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+require 'rails_helper'
+
+RSpec.describe Like, type: :model do
+  describe 'Validations' do
+    let(:user) do
+      User.create(name: 'Rishi', photo: 'Image will be displayed here', bio: 'Hello guys', posts_counter: 0)
```
Comment (TadesseDev) [inline spec/models/like_spec.rb:6]: - [ ] Here Also, Please add a test for the `update_like_count` method inside your `Like` model. We need to make sure all custom methods are tested.

File: spec/models/comment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Tests Comments
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
+  describe 'Validations' do
+    user = User.create(name: 'Rishi', photo: 'Image will be displayed here', bio: 'Hello pals', posts_counter: 0)
+    post = Post.create(title: 'First Post', text: 'This is my first post', author_id: user.id, comments_counter: 0,
+                       likes_counter: 0)
+    subject { described_class.create(text: 'This is my first comment', post:, user:) }
```
Comment (TadesseDev) [inline spec/models/comment_spec.rb]: - [ ] Here Also, Please add a test for the `update_comment_count` method inside your `Comment` model. We need to make sure all custom methods are tested.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @Rishi-Mishra0704 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x]  Correct GitHub Flow  ✅
 - [x]  Good PR title/description  ✅
 - [x]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- [Optional] Dear @Rishi-Mishra0704, I notice some of your changes are already on the `development` branch. This is not the expected behavior, we need to make sure all the changes for the milestone are visible in your pull request. This makes your change visible to the code reviewer before merging it.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by tugsem** [comment]
## STATUS: APPROVED 👍🎯  
Hello @Rishi-Mishra0704 👋 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⭐ 
✔️ Tests for all methods are implemented
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**